### ISSUE TITLE
How to add ISO 8859-1 symbols into XML ?

### ISSUE THREAD

**Issue by dotzhou** [pabigot/pyxb#64]
Dear Peter @pabigot:

Could I ask another question?

How to solve encoding issues for characters as French, or special ISO 8859-1 symbols as "°C" and "±" ?

I tried as below:

```
text = "-80 to +20 °C : ±(0.226- 0.0028)mm"
notes.append(text)
```

Failed, reasonably.

I set up as below:

```
pyxb._InputEncoding = 'ISO-8859-1'
pyxb._OutputEncoding = 'ISO-8859-1'
```

and Tried to output as below:

```
contents = element.toDOM(element_name="geo:element").toprettyxml(indent='    ', 
                                            encoding='ISO-8859-1')
```

I could try convert "ISO-8859-1" to "UTF-8" be encoding as:

```
 text = "80 to +20 &#176;C : &#177;(0.226- 0.0028)mm"
 notes.append(text)
```

However, I believe you definitely have better and more elegant solutions?

Best wishes,

Ted

**Comment by pabigot** [comment]
I don't do a lot with non utf-8 stuff.  You might look at examples/unicode_jp.  It may be that all you need is something like:

    # -*- coding: iso-8859-1 -*-

near the top of the Python file that uses `'-80 to +20 °C : ±(0.226- 0.0028)mm'`.  And if you're using Python 2 ensure you have `unicode` strings not `str` strings.

**Comment by dotzhou** [comment]
Dear Peter @pabigot:

Great thanks for your kind and quick responses.

The example/unicode_jp seems not relevant. 

Python file open can handle "iso-8859-15" characters.

            line = line.encode('utf-8')
            print(line)
           -80 to +20 Â°C : Â±(0.226- 0.0028 x

            line = line.encode('iso-8859-15')
            print(line)
            -80 to +20 °C : ±(0.226- 0.0028 x

           I failed to append those two lines to Pyxb elements


But If change the as 

-80 to +20 &#176;C : &#177;(0.226- 0.0028 x

I can append it to element .

But that probably is not a solution.

How could I hook pyxb using "iso-8859-15" encoding?
I do installed libxml2, probably pyxb still not hooked to libxml2 yet?

import drv_libxml2
import libxml2

################################################################################
logger = logging.getLogger('log2xml')

################################################################################
def setup():
    logging.config.fileConfig('logging.conf')
    pyxb.utils.domutils.BindingDOMSupport.DeclareNamespace(gml.Namespace, 'gml')
    pyxb.utils.domutils.BindingDOMSupport.DeclareNamespace(xlink.Namespace, 'xlink')
    pyxb.utils.domutils.BindingDOMSupport.DeclareNamespace(gmd.Namespace, 'gmd')
    pyxb.utils.domutils.BindingDOMSupport.DeclareNamespace(gco.Namespace, 'gco')
    pyxb.utils.domutils.BindingDOMSupport.DeclareNamespace(geo.Namespace, 'geo')
    pyxb.RequireValidWhenGenerating(True)
    pyxb._InputEncoding = 'iso-8859-15'
    pyxb._OutputEncoding = 'iso-8859-15'
    pyxb.utils.saxutils.SetCreateParserModules(['drv_libxml2'])



I guess what I did is still not enough to hook pyxb to use libxml2?

Also great thanks in advance.

Best wishes,

Ted

**Comment by dotzhou** [comment]
Dear Peter @pabigot:

I get around this issues as below:

        line = line.encode('ascii', 'xmlcharrefreplace')
        print(line)

        -80 to +20 &amp;#194;&amp;#176;C : &amp;#194;&amp;#177;(0.226- 0.0028 x

       Then you can append this line to element.

Probably you or other Pyxb users has better or more elegant solution?

Best wishes,

Ted

**Comment by pabigot** [comment]
I don't know.  It's very hard to answer these without spending an awful lot of time coming up with schema that support the problem and code that lacks only the piece you need.  If you provide a small self-contained schema, the small application code demonstration of what you want to do, and the pyxbgen commands to show what doesn't work I may be able to help.

**Comment by pabigot** [comment]
Absent further information I'm going to close this, assuming it'll be resolved as a side effect of #67.
